### DISCUSSION TITLE
Unable to change the default download directory

### DISCUSSION THREAD

**Discussion by snailbrick** [aclap-dev/video-downloadhelper#380] [Help and Q&A] [answered]
Your question

Hi everyone,

I'm having difficulty selecting a different default download location. When I go to VDH settings, behavior, and "change" next to default download directory, I receive "Error Native port disconnected" I can click cancel, but it results in an endless loop of error native port disconnected messages until I close the tab.

I also receive the same "Error Native port disconnected" if I select Download from the options instead of the quick download buttons.

To clarify, if I use the quick download button, the downloads work fine. I would just like to be able to change the default folder to another location. I have reinstalled VDH and updated the coapp. I'm using Windows 10 64 bit. Sorry if this has been solved previously, but I searched through everything I could find and didn't see someone with the exact same problem. Thanks for the help.

![defaultdirectoryvdh](https://github.com/aclap-dev/video-downloadhelper/assets/166185724/b431d33a-dad8-4b0d-b72c-992a896b7251)
### 

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Ed",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by jf-clavel** [comment]
Windows 64 bits
Je rencontre le même problème depuis le 5/04/2024 , sous firefox version 8.2.2.8 ou sous opera version 8.2.2.9 Suppression, reset,  et réinstallation sont sans effet. Impossible de changer le dossier de téléchargement par défaut. La combobox du choix du dossier ne fonctionne pas.

I have encountered the same problem since 04/5/2024, under Firefox version 8.2.2.8 or under Opera version 8.2.2.9 removing the extension, reset, and reinstallation have no effect.   Impossible to change the default download folder. The folder selection combo box does not work.
![anomalie vdh](https://github.com/aclap-dev/video-downloadhelper/assets/62204834/42ca8de3-2dc1-46a9-8552-c29f5941f52f)

**Reply by warb12** [reply]
I am having the same problem.... reinstalled firefox, installed helper and companion:

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\warb",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by warb12** [comment]
Ok , I found my PATH environment variable was missing C:\Windows\System32 ...

Path Issue: Verify if the PATH environment variable is correctly set. . If the path is not set correctly, you can add it manually:
In Settings search for PATH
choose: Edit environment variable for your account
In the System Variables section, find Path, and click Edit.
Add C:\Windows\System32\ to the list of paths.
Click OK to save the changes.

**Comment by paulrouget** [comment, marked as answer]
In the settings, can you enable Native File Picker. Does it help?

**Reply by snailbrick** [reply]
I tried your suggestion and so far it appears to be working. It opened a normal Windows Folder Selection prompt and I was able to change to another drive. The first download worked fine. I'll write back if I encounter any other issues with the default folder. Thanks a ton!
